codes/extractive/netsum/netsum.py

```diff
@@ -64,15 +78,18 @@ def data_prep(text_path, summary_path, file_info, save_transformed_flag, transfo
             cum_prob_word = 0
             cum_prob_bigram = 0
             word_array = word_tokenizer.tokenize(text_sents[j])
-            all_sent_bigram = list(nltk.bigrams(word_array))
+            word_array_clean = []
             for w in word_array:
-                if w in summary_array:
+                word_array_clean.append(lemmatizer.lemmatize(w,pos='v'))
+            all_sent_bigram = list(nltk.bigrams(word_array_clean))
+            for w in word_array_clean:
```

Comment:
do you consider duplicate words in array? for example as below, your algo will count `and` for 4 times; is it more reasonable to count `and` only 1 time? I saw @chenshaw1995 use the second choice in his evaluation code.

word_array: and, and, and, and, ....
sum_array, and,...

Reply:
So in the original paper, they did not use lemmatization, and that's my very first version. Later I realized that lemmatization did help, as we can have words in different forms. I looked at this piece of code closely and realize that it should be in this way - as `all_docs_words_cleaned` is critical to calculating bigrams. If we only save the non-duplicated version, order will not be preserved correctly and `all_docs_bigram` will fail. But yes, `total_num_words` should consider non-duplicated length.